addons/test_mail/static/tests/attachment_view.test.js

```diff
@@ -0,0 +1,137 @@
+            const doc = popoutIframe.contentDocument;
+            const originalWrite = doc.write;
+            doc.write = (content) => {
+                // This avoids duplicating the test script in the popoutWindow
+                const sanitizedContent = content.replace(
+                    /<script\b[^<]*(?:(?!<\/script>)<[^<]*)*<\/script>/gi,
+                    ""
+                );
+                originalWrite.call(doc, sanitizedContent);
+            };
+            return doc;
+        },
+        get body() {
+            return popoutWindow.document.querySelector(".o_popout_wrapper");
+        },
+        close: () => {
+            popoutWindow.closed = true;
+            popoutIframe.remove(popoutWindow.body);
+        },
+        contains: async (selector) => {
+            await animationFrame();
+            await waitUntil(() => popoutWindow.body);
+            const target = popoutWindow.body.querySelector(selector);
+            expect(target).toBeDisplayed();
+            return target;
+        },
+        click: async (selector) => {
+            const target = await popoutWindow.contains(selector);
+            hootClick(target);
+        },
+    };
+    patchWithCleanup(browser, {
+        open: () => {
+            queryOne(".o_popout_holder").append(popoutIframe);
+            return popoutWindow;
+        },
+    });
+
+    const pyEnv = await startServer();
+    const recordId = pyEnv["mail.test.simple.main.attachment"].create({
+        display_name: "first partner",
+        message_attachment_count: 2,
+    });
+    const attachmentIds = pyEnv["ir.attachment"].create([
+        {
+            mimetype: "image/jpeg",
+            res_id: recordId,
+            res_model: "mail.test.simple.main.attachment",
+        },
+        {
+            mimetype: "application/pdf",
+            res_id: recordId,
+            res_model: "mail.test.simple.main.attachment",
+        },
+    ]);
+    pyEnv["mail.message"].create({
+        attachment_ids: attachmentIds,
+        model: "mail.test.simple.main.attachment",
+        res_id: recordId,
+    });
+    registerArchs({
+        "mail.test.simple.main.attachment,false,form": `
+                <form string="Test document">
+                    <div class="o_popout_holder"/>
+                    <sheet>
+                        <field name="name"/>
+                    </sheet>
+                    <div class="o_attachment_preview"/>
+                    <chatter/>
+                </form>`,
+    });
+
+    patchUiSize({ size: SIZES.XXL });
+    await start();
+    await openFormView("mail.test.simple.main.attachment", recordId);
+    await click(".o_attachment_preview .o_move_next");
+    await contains(".o_attachment_preview img");
+    await click(".o_attachment_preview .o_move_previous");
+    await contains(".o_attachment_preview iframe");
+
+    await click(".o_attachment_preview .o_attachment_control");
+
+    await waitUntil(() => popoutWindow.body);
```

Comment:
Nitpicking: I don't like whitespaces inside scope of test.
The whitespaces help shaping a test suite file, with each test separated by whitespace.
The arbitrary whitespace inside test confused understanding the shape of file, so it's harder to navigate in the file with confidence (e.g. seeing the beginning & end of a given test, even a few wasted seconds is already annoying).

So prefer no whitespaces at all. Ok around inner functions, but otherwise no whitespaces.